### PR TITLE
updated vault helm chart doc with usecase of nlb

### PR DESCRIPTION
**Pull request by rajesht-source**
### Description
This PR updates the document of vault helm about what needs to be done if there is a special requirement of running vault with NLB.

### TODO only if you're a HashiCorp employee
- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Comment by rajesht-source** [comment]
This PR addresses this issue : https://github.com/hashicorp/vault/issues/27592

**Comment by rajesht-source** [comment]
Hi @miagilepner, 
Could you please advise / help  on how this PR can be merged in?  This PR is about to just update the document.

**Comment by rajesht-source** [comment]
@miagilepner could you please

**Comment by rajesht-source** [comment]
Hi @hashishaw Could you please review this PR and advise if this can be approved?

**Comment by rajesht-source** [comment]
Hi @schavis this branch was out of sync with main so I just updated it, it should be good for your review.
Thanks!

File: website/content/docs/platform/k8s/helm/index.mdx [hunk 1]
```diff
@@ -59,6 +59,8 @@ cluster](https://kubernetes.io/docs/tasks/administer-cluster/securing-a-cluster/
 options](/vault/docs/platform/k8s/helm/configuration), and read the [production deployment
 checklist](/vault/docs/platform/k8s/helm/run#architecture).
 
+-> **Important:** This helm chart always provisions Application Load Balancer(ALB) at the AWS side, if you have the requirements of running vault with Network Load Balancer(NLB) then you need to provision NLB in front of your ALB. This special kind of requirement can arise where you need to use AWS PrivateLink with your vault and PrivateLink only works with NLB.
```
Comment (schavis) [inline website/content/docs/platform/k8s/helm/index.mdx:62]:
```suggestion

<Important title="Helm chart provisions ALB on AWS side">

  If you use AWS features (e.g, AWS PrivateLink) that require a network load
  balancer (NLB), you must provision your NLB **before** your application load 
  balancer (ALB).

</Important>
```

Style correction: write in active voice

**Review by schavis — CHANGES_REQUESTED** [review]

**Comment by rajesht-source** [comment]
Hi @schavis I mad ethe suggested change , could you please review and approve?

**Review by schavis — APPROVED** [review]

**Comment by rajesht-source** [comment]
@schavis  Thanks for the approval on this PR, would need one more approval in order to get it merged.
Will really appreciate, if I can get one more approval on this.

Thanks!

**Comment by rajesht-source** [comment]
Hi @schavis, could we please get some of your colleague to review this PR, so that we can merge it in?

**Review by VioletHynes — APPROVED** [review]
Hey there! Thanks for this. Since this is a docs-only change and Sarah's already approved it, I'm going to go ahead and merge this. Thanks for the contribution, and I apologize it took so long to get merged.

**Pull request merged**